Добавить тесты с архитектурой POM для сайта проката самокатов

alenaqa5 opened the pull request:
В данном проекте содержатся:
- allure-report - папка с отчетом по тестам 
(для того чтобы визуально отобразить отчет потребуется команда: allure open allure-report)

- allure-results - папка с детальными результатами запуска тестов
- locators - папка с локаторами для страниц
- pages - папка с описанием страниц
- tests - папка с тестами по страницам FAQ и заказам самоката
- файл conftest.py с фикстурой драйвером.
- файл readme.md, с описанием проекта
- файл requirements.txt - с необходимыми библиотеками

alenaqa5 committed:
добавить тесты с архитектурой POM для сайта проката самокатов

AlexKitIt commented on pages/main_page.py:
```diff
@@ -0,0 +1,47 @@
+    @allure.step('Пролистать страницу до вопроса в разделе "Частые вопросы"')
+    def scroll_to_faq(self, question_key):
+        locator = MainPageLocators.QUESTIONS[question_key]
+        question = self.driver.find_element(*locator)
+        self.driver.execute_script("arguments[0].scrollIntoView();", question)
+        WebDriverWait(self.driver, 10).until(
+            EC.element_to_be_clickable(question)
```
Нужно исправить: здесь и далее по проекту: все прямые обращения к driver и waitы необходимо вынести в base и оттуда за счет наследования переиспользовать в объектах страниц

AlexKitIt commented on tests/test_main_page.py:
```diff
@@ -0,0 +1,26 @@
+    @allure.title('Раскрытие вопросов в разделе "Вопросы о важном"')
+    @allure.description('В разделе "Вопросы о важном" есть список вопросов с ответами, проверяем содержание текста в ответах и функционал раскрытия вопросов')
+    @pytest.mark.parametrize("question_locator, answer_locator, expected_answer", [
+        ("HOW_MUCH_IT_COST", "HOW_MUCH_IT_COST_ANSWER", "Сутки — 400 рублей. Оплата курьеру — наличными или картой."),
+        ("FEW_SCOOTERS", "FEW_SCOOTERS_ANSWER", "Пока что у нас так: один заказ — один самокат. Если хотите покататься с друзьями, можете просто сделать несколько заказов — один за другим."),
+        ("RENT_TIME", "RENT_TIME_ANSWER", "Допустим, вы оформляете заказ на 8 мая. Мы привозим самокат 8 мая в течение дня. Отсчёт времени аренды начинается с момента, когда вы оплатите заказ курьеру. Если мы привезли самокат 8 мая в 20:30, суточная аренда закончится 9 мая в 20:30."),
+        ("ORDER_SCOOTER_FOR_TODAY", "ORDER_SCOOTER_FOR_TODAY_ANSWER", "Только начиная с завтрашнего дня. Но скоро станем расторопнее."),
+        ("PROLONGE_AND_RETURN_SCOOTER", "PROLONGE_AND_RETURN_SCOOTER_ANSWER", "Пока что нет! Но если что-то срочное — всегда можно позвонить в поддержку по красивому номеру 1010."),
+        ("CHARGER_FOR_SCOOTER", "CHARGER_FOR_SCOOTER_ANSWER", "Самокат приезжает к вам с полной зарядкой. Этого хватает на восемь суток — даже если будете кататься без передышек и во сне. Зарядка не понадобится."),
+        ("AVALABILLITY_OF_CANCELATION", "AVALABILLITY_OF_CANCELATION_ANSWER", "Да, пока самокат не привезли. Штрафа не будет, объяснительной записки тоже не попросим. Все же свои."),
+        ("SUBURBS_OF_MOSCOW", "SUBURBS_OF_MOSCOW_ANSWER", "Да, обязательно. Всем самокатов! И Москве, и Московской области."),
```
Можно лучше: тестовые данные стоит хранить отдельно в data модуле, так поддерживать и переиспользовать будет проще

AlexKitIt commented on pages/base_page.py:
```diff
@@ -0,0 +1,37 @@
+base_page_locators = BasePageLocators()
+
+
+class BasePage:
```
Перепутано само назначение BasePage, там не только методы работы с общими элементами (чего по хорошему там тоже не должно быть) но и все низкоуровневые действия без привязки к конкретным элементам

alenaqa5 committed:
Изменить методы в base_page (добавлены методы для взаимодействия с элементами, методы наследуются в дочерние классы и используются в тестах)
Из base_page.py перенести методы для взаимодействия с логотипами.
Из base_page_locators.py перенести локаторы в соответствующие файлы с локаторами для страниц
